repl/src/main/scala/com/cloudera/livy/repl/Session.scala

```diff
@@ -110,18 +116,18 @@ class Session(interpreter: Interpreter, stateChangedCallback: SessionState => Un
 
       info(s"Statement $statementId is cancelling")
 
-      val singleThreadExecutor = ExecutionContext.fromExecutorService(
-        Executors.newSingleThreadExecutor())
       Future {
         if (oldStmtState == StatementState.Running) {
```

Comment:
After moving the cancel logic into a future, I think it makes more sense to use the latest statement state instead of `oldStmtState`.